server.js

```diff
@@ -55,14 +62,20 @@ class Metadata {
                     }).join(' ');
                     doc.scopes = e.scope.split(",");
                 }
-                self.builder.add(doc);
+                if (args.redis) {
+                    this.redis_idx.add(doc);
+                } else {
+                    this.lunr_idx.add(doc);
+                };
             }
             self.db[e.id] = e;
             ++self.count;
         }]);
         self._p.on('data', () => {});
         self._p.on('end', () => {
-             self.index = self.builder.build();
+            if (!args.redis) {
```

Comment:
Again the class approach makes it easy to have a "build" method in the index class.